### ISSUE TITLE
Handle server-interrupted downloads more gracefully

### ISSUE THREAD

**Issue by vijayiyer05** [emeyers/Brain-Observatory-Toolbox#94]
Currently the BOT handles client-interrupted downloads (not marked as cached, overwritten) but not server-interrupted downloads.

When a server error occurs during download, sometimes an .html file is added to file cache alongside a partially downloaded file. This could possibly be used to help detect this situation.

**Comment by vijayiyer05** [comment]
Consider whether [backgroundPool ](https://www.mathworks.com/help/matlab/ref/parallel.backgroundpool.html) may be useful here

**Comment by vijayiyer05** [comment]
@ehennestad It's hard to simulate a server error, so if you think this is sufficiently tested, please move to Done.

**Comment by ehennestad** [comment]
I have experienced a few server errors, but they always caused the websave / matlab.net.http.RequestMessage/send to throw errors. I am therefore not sure how to test the original cause of this issue. 

The extra condition where a file is only added to the cache manifest if the size of the file on the disk matches the size of the file on the web (143e66d6dd4f48cff841c11156988b93366a42c3) I think should be a robust way to make sure that 1) an incomplete file is not added to the cache and 2) the user is notified if the download was incomplete.

As a sidenote, I have added a cleanup function in commit 4a03c9bc7c664fe343e14776bd8fab8e2538fa1d, where, if the download is interrupted by client or server, the partially downloaded file is deleted.
